### PR TITLE
Don't require specific traits or namespaces in scope for the glib_wra…

### PR DESCRIPTION
**Pull request by sdroege**
…pper! macro

This adds various unused use compiler warnings everywhere, and needs a gir fix for removing those. See https://github.com/gtk-rs/gir/pull/671

### CODE REVIEW

File: src/boxed.rs [hunk 1]
```diff
@@ -214,14 +214,14 @@ macro_rules! glib_boxed_wrapper {
...
-                    res.push($crate::translate::from_glib_none(ptr::read(ptr.add(i))));
+                    res.push($crate::translate::from_glib_none(::std::ptr::read(ptr.add(i))));
                 }
                 res
             }
 
             unsafe fn from_glib_container_num_as_vec(ptr: *mut *mut $ffi_name, num: usize) -> Vec<Self> {
                 let res = $crate::translate::FromGlibContainerAsVec::from_glib_none_num_as_vec(ptr, num);
-                glib_ffi::g_free(ptr as *mut _);
+                $crate::ffi::g_free(ptr as *mut _);
```
Comment (EPashkin) [inline src/boxed.rs]: Seems this not works in other crates

Comment (EPashkin) [reply src/boxed.rs]: See appveyour logs, I don't think that it fails in `ffi::g_tls_file_database_get_type(),`.
```
error[E0603]: module `ffi` is private
  --> C:\Users\appveyor\.cargo\git\checkouts\gio-a4b825b310a073b9\d1f28c4\src\auto\tls_file_database.rs:25:1
   |
25 | / glib_wrapper! {
26 | |     pub struct TlsFileDatabase(Object<ffi::GTlsFileDatabase, ffi::GTlsFileDatabaseInterface>): TlsDatabase;
27 | |
28 | |     match fn {
29 | |         get_type => || ffi::g_tls_file_database_get_type(),
30 | |     }
31 | | }
   | |_^
   |
   = note: this error originates in a macro outside of the current crate (in Nightly builds, run with -Z external-macro-backtrace for more info)
```
also many errors in travis but its don't show from here it
```
error[E0603]: extern crate `ffi` is private
  --> <::glib::boxed::glib_boxed_wrapper macros>:78:24
   |
78 | let v_ptr = $ crate :: ffi :: g_malloc0 (
   |                        ^^^
```

Comment (EPashkin) [reply src/boxed.rs]: .. so my assumption that `extern crate glib_sys as ffi;` don't create exported module

Comment (sdroege) [reply src/boxed.rs]: Ah right. This needs the boxed PR. That one is reexporting ffi

Comment (EPashkin) [reply src/boxed.rs]: I missed that you added `pub` to it and didn't understand why `#[doc(hidden)]` added

Comment (EPashkin) [reply src/boxed.rs]: .. now it make sense.
Then we wait until #397 merged

**Comment by GuillaumeGomez** [comment]
I restarted CIs.

**Comment by sdroege** [comment]
@GuillaumeGomez all green, we can go ahead with this one at least :) gir fixes come afterwards to get rid of all the compiler warnings. Will be easier to catch them all once this is merged

**Comment by GuillaumeGomez** [comment]
Indeed, thanks!

**Pull request merged**